src/utils/graphicsutils.cpp

```diff
@@ -73,51 +75,30 @@ void GraphicsUtils::distanceFromLine(double cx, double cy, double ax, double ay,
 struct PD {
 	QPointF p;
 	double d;
-};
+    PD(qreal x, qreal y, double _d) : p(x, y), d(_d) { }
 
-bool pdLessThan(PD* pd1, PD* pd2) {
-	return pd1->d < pd2->d;
+};
+inline bool operator<(const PD& lhs, const PD& rhs) noexcept {
+    return lhs.d < rhs.d;
 }
 
 QPointF GraphicsUtils::calcConstraint(QPointF initial, QPointF current) {
-	QList<PD *> pds;
-
-	PD * pd = new PD;
-	pd->p.setX(current.x());
-	pd->p.setY(initial.y());
-	pd->d = (current.y() - initial.y()) * (current.y() - initial.y());
-	pds.append(pd);
-
-	pd = new PD;
-	pd->p.setX(initial.x());
-	pd->p.setY(current.y());
-	pd->d = (current.x() - initial.x()) * (current.x() - initial.x());
-	pds.append(pd);
-
+    std::vector<PD> pds;
```

Comment:
I would have used QList but it does not provide the ability to construct objects in place.